### PR TITLE
Generated files move: app/common/gen movement to zzz_generated

### PR DESCRIPTION
**Pull request by andy31415**
#### Problem
Want to decouple generated file paths from strict location inside the source directory.

#### Change overview
Moves app/common/gen into zzz_generated. paths changed from includes of `app/common/gen` into `app-common/zap-generated`

This will make dependencies visible within source code (including some odd dependencies, like data-model and source includes) as a sideffect, which will allow us to clean up how generated code is organized and try to make it more independent.

#### Testing

Locally compliled all that build_examples.py  can build as well as xcode project compilation. The rest should be CI (CI compilation should pass).

### CODE REVIEW

**Review by LuDuda — COMMENTED** [review]
Could you please update the [.gitattributes](https://github.com/project-chip/connectedhomeip/blob/master/.gitattributes) file?

**Comment by andy31415** [comment]
> Could you please update the [.gitattributes](https://github.com/project-chip/connectedhomeip/blob/master/.gitattributes) file?

Done, was not aware that was a thing (should have been done in previous PR). Checked in the restyle and gitattributes change.

**Comment by andy31415** [comment]
Reviewer notes for interesting changes:
  - scripts/tools/zap_regen_all.py I started adding the option to specify output directory for what we generate. Used it only in  one place (app/common) however I would like to eventually change all of these
  - top-level includes stopped working:
     - Chaged src/BUILD.gn to add zzz_generated  as a top level include (I do not like this, should be changed in time, I think PR would become to big trying to also fix this)
     - Changed src/app/chip_data_model.gni  to allow extra include_dirs arguments, because src/controller/data_model/BUILD.gn (data_model depends on app includes). Unsure if this is really needed because the src/BUILD.gn change, however it seemed  an explicit dependency to call out, so thinking to keep even if just as a flag of 'we have  generated files depending on other generated files'.
  -  Lots of cmakefiles changes to do the same: add the zzz_generated include path to be visible  during compilation since gn is not used there.

**Review by woody-apple — APPROVED** [review]

File: src/controller/data_model/BUILD.gn [hunk 1]
```diff
@@ -23,4 +23,7 @@ chip_data_model("data_model") {
 
   use_default_client_callbacks = true
   allow_circular_includes_from = [ "${chip_root}/src/controller" ]
+
+  # TODO: generated file dependency should be a lib NOT an include
+  include_dirs = [ "${chip_root}/zzz_generated/app-common" ]
```
Comment (mspang) [inline src/controller/data_model/BUILD.gn:28]: Use the one from src/ ?

Comment (andy31415) [reply src/controller/data_model/BUILD.gn]: Done, removed the include_dirs support.

This was remaining from where I found that at least data_model has cross dependencies (all examples gen files have this too) and never removed it thinking 'good marker'. However this is easily changed when we fix all dependencies anyway.

**Review by msandstedt — APPROVED** [review]

**Review by mspang — APPROVED** [review]

**Pull request merged**